lib/net.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Net - abstraction layer around Node's HTTP/S request system.
+ * Advantages:
+ * - easier acquiring of data
+ * - mass disabling of outgoing requests via Config.
+ */
+
+import * as https from 'https';
+import * as http from 'http';
+import * as url from 'url';
+import * as Streams from './streams';
+
+export class URIRequest {
+	uri: string;
+	statusCode?: number;
+	constructor(uri: string) {
+		this.uri = uri;
+		this.statusCode = undefined;
+	}
+	/**
+	 * Makes a basic http/https request to the URI.
+	 * Returns the response data.
+	 * @param opts request opts - headers, etc.
+	 */
+	async get(opts?: AnyObject): Promise<string | null> {
+		if (Config.noURIRequests) return null;
```

Comment:
I would also call this `noNetRequests`.

Maybe this should throw instead of returning `null`? Would make it easier to use in practice.